src/main/java/com/epam/ta/reportportal/core/imprt/impl/junit/AsyncJunitImportLaunch.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 EPAM Systems
+ *
+ *
+ * This file is part of EPAM Report Portal.
+ * https://github.com/reportportal/service-api
+ *
+ * Report Portal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Report Portal is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.epam.ta.reportportal.core.imprt.impl.junit;
+
+import com.epam.ta.reportportal.core.imprt.impl.ImportLaunch;
+import com.epam.ta.reportportal.core.launch.IFinishLaunchHandler;
+import com.epam.ta.reportportal.core.launch.IStartLaunchHandler;
+import com.epam.ta.reportportal.database.dao.LaunchRepository;
+import com.epam.ta.reportportal.database.entity.Launch;
+import com.epam.ta.reportportal.exception.ReportPortalException;
+import com.epam.ta.reportportal.ws.model.ErrorType;
+import com.epam.ta.reportportal.ws.model.FinishExecutionRQ;
+import com.epam.ta.reportportal.ws.model.launch.Mode;
+import com.epam.ta.reportportal.ws.model.launch.StartLaunchRQ;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.inject.Provider;
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.concurrent.*;
+import java.util.function.Predicate;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+@Service
+public class AsyncJunitImportLaunch implements ImportLaunch {
+
+    @Autowired
+    private Provider<JunitParseJob> xmlParseJobProvider;
```

Comment:
yes, cause if we go further into JunitParseJob we will find autowired provider for sax handler, and sax handler also contains autowired services